### ISSUE TITLE
2D Views:  Double Partition measurements not overlapping

### ISSUE THREAD

**Issue by TylerStandage** [CCDevNelson/Snap-Updates#324]
Sometimes when there are double partitions, the measurements of partition heights overlap as in this image: 

![image](https://user-images.githubusercontent.com/77167318/110818990-9f7aa700-8242-11eb-8efa-a467068139a0.png)

We should move the measurements of the partition on the right side to inside the opening it pertains to like below:

![image](https://user-images.githubusercontent.com/77167318/110819170-c638dd80-8242-11eb-8891-c2b8c4dae023.png)

Even better if we can flip the numbers on the right side to have the "bottom" side of the numbers butting up against the partition it pertains to. 

Let me or @sarahhightower know if you have questions.

**Comment by nailsonlandim** [comment]
Hi @TylerStandage and @sarahhightower,

Now the partition height labeling checks for overlaps and displaces the dimensions, one at 1/4 of the partition height and the other at 3/4 of the partition height. It can handle an overlapping tolerance of one inch. above that, I think it won't overlap.

![324_1](https://user-images.githubusercontent.com/70038700/111328191-c96e0800-864c-11eb-8689-20af5868708a.png)

Tell me what you think. It's available at: https://github.com/nailsonlandim/SNaP-Dev/tree/324-double-partition-overlapping

**Comment by TylerStandage** [comment]
Thanks @nailsonlandim, great that it doesn't overlap, which is most important.  I had suggested we also move the numbers for the pards to inside the section they pertain to, possible flipping them 180 degrees to rest against the relevant pard.

@sarahhightower are you ok with the approach nailson has taken?  It does seem to avoid the overlap, which may be sufficient.

**Comment by nailsonlandim** [comment]
>I had suggested we also move the numbers for the pards to inside the section they pertain to, possible flipping them 180 degrees to rest against the relevant pard.

@TylerStandage I'll check with @AntonioGloria about this possibility. Now that they are detected and handled we can perform experiments.

**Comment by TylerStandage** [comment]
@nailsonlandim let's wait to dig into it more until we get sarah's opinion.  If this is good enough, then we can call it good.

**Comment by TedCC1209** [comment]
I'd much prefer Tyler's suggestion of moving the dimension to the inside of the section of the pertain to, with or without flipping it 180 degrees.

**Comment by TylerStandage** [comment]
@nailsonlandim then let's go ahead and figure out how to do it.  thanks for the weigh-in @TedCC1209

**Comment by sarahhightower** [comment]
> Thanks @nailsonlandim, great that it doesn't overlap, which is most important. I had suggested we also move the numbers for the pards to inside the section they pertain to, possible flipping them 180 degrees to rest against the relevant pard.
> 
> @sarahhightower are you ok with the approach nailson has taken? It does seem to avoid the overlap, which may be sufficient.

I'd really like it on the inside as well. Thanks!

**Comment by TylerStandage** [comment]
Note that putting it inside the other section is only needed when there is a double pard... Not sure if that makes it harder to find a solution for. Please let me know if you have questions or want to talk about what it should look like.

**Comment by nailsonlandim** [comment]
Hi @TylerStandage @TedCC1209 and @sarahhightower,

I did the adjustments.

![image](https://user-images.githubusercontent.com/70038700/111384344-24712080-8688-11eb-8d77-0a3933e7920a.png)

Tell me what you think. Available at: https://github.com/nailsonlandim/SNaP-Dev/tree/324-double-partition-overlapping

**Comment by TylerStandage** [comment]
@nailsonlandim this looks good to me! @sarahhightower @TedCC1209 any further comments based on the above?

**Comment by TedCC1209** [comment]
Thanks Nailson. Can we have them still be centered vertically on the partition? 

![double pard labels](https://user-images.githubusercontent.com/61025894/111387072-655a3c80-866a-11eb-8951-aaf3a9d4ef11.png)

**Comment by nailsonlandim** [comment]
> Thanks Nailson. Can we have them still be centered vertically on the partition?

I can tweak a bit more to make it as similar as possible, but rotation on labels works in a different way than other objects, 

The dimension upwards (90 degrees)
![image](https://user-images.githubusercontent.com/70038700/111388674-7a48c700-868e-11eb-953f-94d1361cce0d.png)

The same dimension tilted 180 degrees (-90 degrees)
![image](https://user-images.githubusercontent.com/70038700/111388749-9c424980-868e-11eb-8d36-007c6c8727da.png)

And at 0 degrees
![image](https://user-images.githubusercontent.com/70038700/111388773-a8c6a200-868e-11eb-9aed-767bf6fda7ea.png)

The text for standard dimensions gets centered only on 0 degrees.

**Comment by TedCC1209** [comment]
Personally I'd prefer it be centered and not rotated rather than rotated and off-center. Not sure what @TylerStandage and @sarahhightower think though.

**Comment by TylerStandage** [comment]
No strong opinion on it.  @sarahhightower ?

**Comment by sarahhightower** [comment]
I agree with @TedCC1209. I would prefer it being centered over it being rotated.

**Comment by nailsonlandim** [comment]
Agreed, it looks much better now. Thanks @TylerStandage @TedCC1209 and @sarahhightower for the prompt feedback (and patience)

![image](https://user-images.githubusercontent.com/70038700/111391357-360bf580-8693-11eb-8921-bfe88f44a3b1.png)

**Comment by TylerStandage** [comment]
I'd call that good to go, thanks @nailsonlandim !
